File: smartnote_core/src/models/note.rs
//! Module contains Note struct
use std::path::PathBuf;

#[derive(serde::Serialize, serde::Deserialize)]
pub struct Note {
    #[serde(with = "uuid::serde::simple")]
    pub local_id: uuid::Uuid,
    pub mongo_id: Option<String>,
    #[serde(with = "uuid::serde::simple")]
    pub owner_id: uuid::Uuid,

    pub name: String,
    pub title: String,
    pub summary: String,
    pub content_path: PathBuf,

    pub created_at: i64,
    pub updated_at: i64,
    pub is_deleted: bool,
    pub deleted_at: Option<i64>,

    pub version: i64, //zmieniać wersje o 1 co zapis, porównywać na czasie ostatniej modyfikacji, wersja tylko pomocniczo
    pub cloud_version: Option<i64>,
    pub sync_state: crate::services::storage::db_creation::SyncState,

    pub encrypted: bool,
    pub crypto_meta: Option<serde_json::Value>,
}

File: desktop/src/types/note.ts
export interface Note {
  local_id: string;
  mongo_id: string | null;
  owner_id: string;

  title: string;
  summary: string;
  content_path: string;

  created_at: number;
  updated_at: number;
  is_deleted: boolean;
  deleted_at: number | null;

  version: number;
  cloud_version: number | null;
  sync_state: SyncState;

  encrypted: boolean;

  crypto_meta: any | null;
}

export type SyncState =
  | 'Synced'
  | 'PendingUpload'
  | 'PendingDownload'
  | 'Conflict'
  | 'Error';